src/dodal/devices/pressure_jump_cell.py

```diff
@@ -0,0 +1,240 @@
+
+
+class ValveState(str, Enum):
+    FAULT = "Fault"
+    OPEN = "Open"
+    OPENING = "Opening"
+    CLOSED = "Closed"
+    CLOSING = "Closing"
+
+
+class FastValveState(str, Enum):
+    FAULT = "Fault"
+    OPEN = "Open"
+    OPEN_ARMED = "Open Armed"
+    CLOSED = "Closed"
+    CLOSED_ARMED = "Closed Armed"
+    NONE5 = "Unused"
+
+
+class LimitSwitchState(str, Enum):
+    OFF = "Off"
+    ON = "On"
+
+
+class AllValvesControl(StandardReadable):
+    """
+    valves 2, 4, 7, 8 are not controlled by the IOC,
+    as they are under manual control.
+    """
+
+    def __init__(self, prefix: str, name: str = "") -> None:
+        with self.add_children_as_readables():
+            self.valve_states: DeviceVector[SignalR[ValveState]] = DeviceVector(
+                {i: epics_signal_r(ValveState, f"{prefix}V{i}:STA") for i in [1, 3]}
+            )
+            self.fast_valve_states: DeviceVector[SignalR[FastValveState]] = (
+                DeviceVector(
+                    {
+                        i: epics_signal_r(FastValveState, f"{prefix}V{i}:STA")
+                        for i in [5, 6]
+                    }
+                )
+            )
+
+        with self.add_children_as_readables(ConfigSignal):
+            self.valves_open: DeviceVector[SignalR[int]] = DeviceVector(
+                {i: epics_signal_rw(int, f"{prefix}V{i}:OPENSEQ") for i in [1, 3, 5, 6]}
+            )
+
+            self.valve_control: DeviceVector[SignalR[ValveControlRequest]] = (
+                DeviceVector(
+                    {
+                        i: epics_signal_rw(ValveControlRequest, f"{prefix}V{i}:CON")
+                        for i in [1, 3]
+                    }
+                )
+            )
+
+            self.fast_valve_control: DeviceVector[SignalR[FastValveControlRequest]] = (
+                DeviceVector(
+                    {
+                        i: epics_signal_rw(FastValveControlRequest, f"{prefix}V{i}:CON")
+                        for i in [5, 6]
+                    }
+                )
+            )
+
+        super().__init__(name)
+
+
+class Pump(StandardReadable):
+    def __init__(self, prefix: str, name: str = "") -> None:
+        with self.add_children_as_readables():
+            self.pump_position = epics_signal_r(float, prefix + "POS")
+            self.pump_forward_limit = epics_signal_r(
+                LimitSwitchState, prefix + "D74IN1"
+            )
+            self.pump_backward_limit = epics_signal_r(
+                LimitSwitchState, prefix + "D74IN0"
+            )
+            self.pump_motor_direction = epics_signal_r(
+                PumpMotorDirectionState, prefix + "MTRDIR"
+            )
+            self.pump_speed_rbv = epics_signal_r(float, prefix + "MSPEED_RBV")
+
+        with self.add_children_as_readables(ConfigSignal):
+            self.pump_mode = epics_signal_rw(PumpState, prefix + "SP:AUTO")
+            self.pump_speed = epics_signal_rw(float, prefix + "MSPEED")
```

Comment:
While the speed, direction etc. seem likely to change within a scan? It's something that could easily be scanned over, e.g. transmission vs. pump speed if the fluid is being rapidly damaged by the beam.

Reply:
so you mean that they shouldn't be added with ConfigSignal?

Reply:
Read signals are read at every point of the scan- if something is changing, or variable or may want to be scanned over, it should be a Read signal.

Config signals are read only at the start of the scan. Something like limits would ideally be metadata on a signal that propagates from the Epics record, but if the data is as additional signals then having it read only once per scan is appropriate.

IMO:

PressureTransducer - perfect

PressureJumpCellController: some of these signals (e.g. GOTOBUSY, TIMER, COUNTER, RESULT, METHOD, ITERATION) seem to be more relevant for the control system itself rather than for reading out the data? I don't know enough about the usage of the device to know which should be read and how often.

pump I think the following is sensible?

```
pump:
        readables:
            self.pump_position = epics_signal_r(float, prefix + "POS")
            self.pump_motor_direction = epics_signal_r(
                PumpMotorDirectionState, prefix + "MTRDIR"
            )
use the epics_signal_rw with different set/read prefixes to have one signal that does both: sets go to set, reads come from _rbv
            self.pump_speed = epics_signal_rw(float, prefix + "MSPEED")

        config:
            self.pump_forward_limit = epics_signal_r(
                LimitSwitchState, prefix + "D74IN1"
            )
            self.pump_backward_limit = epics_signal_r(
                LimitSwitchState, prefix + "D74IN0"
            )
            self.pump_mode = epics_signal_rw(PumpState, prefix + "SP:AUTO")
this is int but the next is bool? Are they just saying which direction the pump is moving? If so, duplicating work of pump_motor_direction?
            self.pump_move_forward = epics_signal_rw(int, prefix + "M1:FORW")
            self.pump_move_backward = epics_signal_rw(bool, prefix + "M1:BACKW")

        super().__init__(name)
```

Reply:
sorted when reworking with at p38 with @DiamondJoseph  on the 21st of August